source/mesh_refinement/nonadiabatic_temperature.cc

```diff
@@ -49,7 +49,7 @@ namespace aspect
                                                   this->get_mpi_communicator());
 
       const Quadrature<dim> quadrature(this->get_fe().base_element(this->introspection().base_elements.temperature).get_unit_support_points());
-      std::vector<unsigned int> local_dof_indices (this->get_fe().dofs_per_cell);
+      std::vector<types::global_dof_index> local_dof_indices (this->get_fe().dofs_per_cell);
       FEValues<dim> fe_values (this->get_mapping(),
```

Comment:
These changes are actually already in master (this comparison apparently is to an older master due to merge conflicts). I cherry-picked the changes into this branch, because I wanted to test the 64bit functionality for the tracers. This should disappear when I rebase (after I went through all of your comments).